src/actions/posts.js

```diff
@@ -759,6 +760,28 @@ export function getPostsAround(channelId, postId, perPage = Posts.POST_CHUNK_SIZ
     };
 }
 
+// getThreadsForPosts is intended for an array of posts that have been batched
+// (see the actions/websocket_actions/handleNewPostEvents function in the webapp)
+export function getThreadsForPosts(posts, dispatch, getState) {
+    if (!Array.isArray(posts) || posts.length === 0) {
+        return Promise.resolve();
+    }
+
+    const promises = [];
+
+    posts.forEach((post) => {
```

Comment:
Aha, here's the complexity. Is it true that for each post we would have "normally" received, we call `getPostThread`, and so doing them all now isn't any "worse" as such? Slightly worried about slamming the server in this scenario, and wondering if we need a `getPostsThreads` instead.

Reply:
Yep, for every normal post we would do this (and more). Here, we only get the thread if we don't have it in the store already. I think a `getPostsThreads` wouldn't be any different (we've got to get each of those threads somehow, right?).

When Christopher was making the rate-limiting changes, he said the performance hit from `getProfilesAndStatusesForPosts` isn't significant, and it's got a lot more going on than `getThreadsForPosts`.  (https://github.com/mattermost/mattermost-webapp/pull/2041)